src/node_file.cc

```diff
@@ -2106,6 +2106,51 @@ static void OpenFileHandle(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+// TODO(@anonrig): Implement v8 fast APi calls for `cpSync`.
+static void CpSync(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args.Length() ==
+        8);  // src, dest, preserveTimestamps, dereference, errorOnExist, force,
+             // recursive, verbatimSymlinks
+  BufferValue src(env->isolate(), args[0]);
+  CHECK_NOT_NULL(*src);
+  ToNamespacedPath(env, &src);
+
+  BufferValue dest(env->isolate(), args[1]);
+  CHECK_NOT_NULL(*dest);
+  ToNamespacedPath(env, &dest);
+
+  bool preserveTimestamps = args[2]->IsTrue();
+  bool dereference = args[3]->IsTrue();
+  bool errorOnExist = args[4]->IsTrue();
+  bool force = args[5]->IsTrue();
+  bool recursive = args[6]->IsTrue();
+  bool verbatimSymlinks = args[7]->IsTrue();
+
+  if (errorOnExist) {
+    if (std::filesystem::exists(src.ToStringView())) {
```

Comment:
This can throw right? Might need to wrap these in a try/catch? https://en.cppreference.com/w/cpp/filesystem/exists

Reply:
Yes, or we can add a second parameter of error code and avoid try/catch.